react-ui/src/pages/entry/EntryModal.jsx

```diff
@@ -179,9 +177,16 @@ function EntryModal() {
         redditCommentButtonRef.current.click();
         return;
       case 'c':
-        // clear vote?
+        clearRating();
         return;
       default:
+        if (key >= '0' && key <= '5') {
+          // ??? I would love it if we could track whether votes mobile or desktop
+          // ??? and if desktop, how many were keyed
+          // https://stackoverflow.com/questions/39435395/reactjs-how-to-determine-if-the-application-is-being-viewed-on-mobile-or-deskto
+          changeRating(parseInt(key, 10), entry?.rating, true);
+          indexChange = 1;
+        }
         return;
```

Comment:
If you want the `indexChange` to be processed, you need a `break` instead of `returning`. I'm a bit worried that errors would be hard to discern and votes might be missed.

My suggestion is change the `changeRating` and `clearRating` functions in `useVoting.jsx` to `async`. And then here:
```js
  const handleKeyedVote = async (rating) => {
    await changeRating(rating, entry?.rating, true);
    handleNavigate(1);
  };

    switch (key) {
      // existing switch cases
      default:
        if (key >= '0' && key <= '5') {
          handleKeyedVote(parseInt(key, 10));
        }
        return; // Note that I've kept return here, because I don't want to handle navigate here and I can't await in an event handler (that I know of).
    }
```

Reply:
Ah ok, so the main issue is that voting often seems to be a little slow / delayed. So there's three options:
1. Auto-navigate asynchronously, but a) the snackbar will appear on the next flag and b) you can still vote for the next flag, which if performed before the previous call completes seem to cancel it out... not sure why
2. Auto-navigate synchronously, but then the delay could lead to key spamming
3. Don't auto-navigate, which may be the best intermediate step for now and we can revisit the asynchronous failure later.

Reply:
I'm fine with 3 for now since it looks like there's a couple of issues to work through.